lib/grammers-mtproto/src/transport/mod.rs

```diff
@@ -91,8 +93,17 @@ pub trait Transport {
     fn pack(&mut self, buffer: &mut DequeBuffer<u8>);
 
     /// Unpacks the input buffer in-place.
-    fn unpack(&mut self, buffer: &[u8]) -> Result<UnpackedOffset, Error>;
+    /// Subsequent calls to `unpack` should be made with the same buffer,
+    /// with the data on the ranges from previous `UnpackedOffset` removed.
+    fn unpack(&mut self, buffer: &mut [u8]) -> Result<UnpackedOffset, Error>;
```

Comment:
There should probably be an explainer on why the `buffer` is `mut` here... As I was wondering.

Reply:
`pack` takes a mutable input so in my view it's reasonable for `unpack` to do so too.

Reply:
By why would we alter the buffer itself? Isn't it supposed to be the read buffer?

Reply:
Both packing unpacking occur in-place. None of the previous transports had to modify the buffer during unpack, but obfuscated does. The caller doesn't read from the buffer contents until the calls succeed.

Reply:
(Forgot to add, but that said, I'm happy to have the documentation improved to leave this in a better place than a random GitHub discussion.)